SignalMessaging/ViewModels/ContactShareViewModel.swift

```diff
@@ -59,6 +59,7 @@ public class ContactShareViewModel: NSObject {
         let avatarBuilder = OWSContactAvatarBuilder(nonSignalName: displayName,
                                                     colorSeed: colorSeed,
                                                     diameter: UInt(diameter),
+                                                    ignoreContactAndProfile: ignoreContactAndProfile,
                                                     contactsManager: contactsManager)
         return avatarBuilder.build()
```

Comment:
So we re-used our existing avatar builder, and have found that it doesn't do *quite* what we need it to do in this circumstance.

Adding a "mode" flag with a default is a quick fix which doesn't alter other code paths, but at the cost of accruing more complexity in the codebase.

What if instead, we revert all the `ignoreContactAndProfile` mode stuff, and instead here we call `avatarBuilder.buildDefaultImage()` directly?